server/admin/admin_server.go

```diff
@@ -139,19 +137,19 @@ func (s *Server) CreateTree(ctx context.Context, request *trillian.CreateTreeReq
 
 	// Derive the public key that corresponds to the private key for this tree.
 	// The caller may have provided the public key, but for safety we shouldn't rely on it being correct.
-	publicKeyDER, err := x509.MarshalPKIXPublicKey(signer.Public())
+	publicKey, err := keys.ToPublicKeyPB(signer.Public())
 	if err != nil {
-		return nil, status.Errorf(codes.InvalidArgument, "failed to marshal public key: %v", err.Error())
+		return nil, status.Errorf(codes.InvalidArgument, "%v", err.Error())
```

Comment:
We probably missed that on the previous reviews, but we don't need `Errorf` if a printf-like pattern is not being used. Up to you whether to consider this out of scope, in which case I can do the Errorf/Error changes in a separate PR.